src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * This file is part of contao-community-alliance/dc-general.
+ *
+ * (c) 2013-2017 Contao Community Alliance.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * This project is provided in good faith and hope to be usable by anyone.
+ *
+ * @package    contao-community-alliance/dc-general
+ * @author     Sven Baumann <baumann.sv@gmail.com>
+ * @copyright  2013-2017 Contao Community Alliance.
+ * @license    https://github.com/contao-community-alliance/dc-general/blob/master/LICENSE LGPL-3.0
+ * @filesource
+ */
+
+namespace ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties;
+
+use ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\Collection\SubPropertyInterface;
+use ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralInvalidArgumentException;
+
+/**
+ * Default implementation of a property with collection definition.
+ */
+class DefaultCompoundProperty extends DefaultProperty implements CompoundPropertyInterface
+{
+    /**
+     * The collection of internal collection property in this property.
+     *
+     * @var array|SubPropertyInterface[]
+     */
+    private $collection = array();
+
+    /**
+     * {@inheritDoc}
+     */
+    public function clearProperties()
+    {
+        $this->collection = array();
+
+        return $this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getProperties()
+    {
+        return $this->collection;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setProperties(array $properties)
+    {
+        $this->clearProperties();
+
+        foreach ($properties as $collectionProperty) {
+            $this->addProperty($collectionProperty);
+        }
+
+        return $this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function hasProperty($name)
+    {
+        return array_key_exists($name, $this->collection);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getProperty($name)
+    {
+        if (!$this->hasProperty($name)) {
+            throw new DcGeneralInvalidArgumentException('Property ' . $name . ' is not registered.');
+        }
+
+        return $this->collection[$name];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function addProperty(SubPropertyInterface $property)
+    {
+        $this->collection[$property->getName()] = $property;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function removeProperty(SubPropertyInterface $property)
+    {
+        $this->collection = array_diff_key($this->collection, array_flip(array($property->getName())));
```

Comment:
Can be simply `unset($this->collection[$property->getName()]);`.

Reply:
change